src/utils/config_utils/src/config_utils.c

```diff
@@ -249,7 +249,11 @@ bool ADUC_Json_GetAgents(JSON_Value* root_value, size_t* agentCount, ADUC_AgentI
         goto done;
     }
 
-    ADUC_ALLOC_BLOCK(*agents, agents_count, sizeof(ADUC_AgentInfo));
+    *agents = calloc(agents_count, sizeof(ADUC_AgentInfo));
```

Comment:
at line 284, we should pass agents_count insted of *agentCount? (*agentCount wont' get assigned if we goto done at line 255)

Reply:
*agentCount is not set until 6 lines down (so *agentCount here would be 0)--we don't want to set that out param until prove that we can allocate the memory.  The variable names are too similar, so renaming agents_count (Note the underscore char) to `json_array_agents_count` here.

Reply:
revamped agents json array parsing based on feedback and made it only return single element (1st element) but log warning if array size is greater than 1